server/stream.go

```diff
@@ -778,7 +782,8 @@ func (a *Account) addStreamWithAssignment(config *StreamConfig, fsConfig *FileSt
 		}
 	}
 	fsCfg.StoreDir = storeDir
-	fsCfg.AsyncFlush = false
+	// Async flushing is only allowed if the stream has a sync log backing it.
```

Comment:
Could we allow for R1s with caveat data loss might be possible?

Reply:
Would like to discuss that, but (if we agree) allow that in a separate PR.

I'm thinking people would likely jump to opt-in to incredible throughput at the cost of consistency for R1, to then ask why writes are lost during hard kills, etc. So, I'm honestly not sure yet if we should be allowing this at all.